### PR TITLE
contracts: Add `FeeCurrencyWhitelist`

### PR DESCRIPTION
**Pull request by palango**
Adds the `FeeCurrencyWhitelist` contract, part of #57.

@pahor167 Please have a look at this. The changes I did:
- Add the license modifier
- Update solidity pragma
- Fix imports
- Removed the `public` visibility modifier from the constructor, because solc complained.

### CODE REVIEW

**Comment by palango** [comment]
Mhg, I'm not sure why [this](https://app.circleci.com/pipelines/github/celo-org/optimism/406/workflows/6812cbf6-4966-4e60-88a3-06bb7d3e8795/jobs/5325) complains about the ABIs. I'm using the `abigen` version that is required:
```
% abigen --version
abigen version 1.10.25-stable
```

**Comment by pahor167** [comment]
> Mhg, I'm not sure why [this](https://app.circleci.com/pipelines/github/celo-org/optimism/406/workflows/6812cbf6-4966-4e60-88a3-06bb7d3e8795/jobs/5325) complains about the ABIs. I'm using the `abigen` version that is required:
> 
> ```
> % abigen --version
> abigen version 1.10.25-stable
> ```

Did you run `make` in op-bindings folder after all changes to smart contract were done ? I took the liberty to fetch branch, rebuild bindings and it seems that op-bindings-build is passing.

`abigen --version
abigen version 1.10.25-stable`

My abigen version is same as yours.

File: op-bindings/predeploys/addresses.go [hunk 1]
```diff
@@ -90,6 +92,7 @@ func init() {
 	Predeploys["EAS"] = &EASAddr
 	Predeploys["CeloRegistry"] = &CeloRegistryAddr
 	Predeploys["GoldToken"] = &GoldTokenAddr
+	Predeploys["FeeCurrencyWhitelist"] = &FeeCurrencyWhitelistAddr
 
 	CeloPredeploys[Predeploys["CeloRegistry"].String()] = Predeploys["CeloRegistry"]
```
Comment (pahor167) [inline op-bindings/predeploys/addresses.go:97]: It is necessary to also add FeeCurrencyWhitelist into CeloPredeploys

**Comment by palango** [comment]
This is included in #70

**Pull request closed without merging**